jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -290,16 +326,35 @@ protected void proceed()
 
     protected Connection activate()
     {
-        Pool<Connection>.Entry entry = pool.acquire();
-        if (entry != null)
+        while (true)
         {
-            if (LOG.isDebugEnabled())
-                LOG.debug("Activated {} {}", entry, pool);
-            Connection connection = entry.getPooled();
-            acquired(connection);
-            return connection;
+            Pool<Connection>.Entry entry = pool.acquire();
+            if (entry != null)
+            {
+                Connection connection = entry.getPooled();
+
+                long maxDurationNanos = this.maxDurationNanos;
+                if (maxDurationNanos > 0L)
+                {
+                    EntryHolder holder = (EntryHolder)((Attachable)connection).getAttachment();
+                    if (holder != null && holder.isExpired(maxDurationNanos))
```

Comment:
No, it can't. The only time the attachment can be null is before the entry is enabled or after it is removed so `pool.acquire()` can never return a connection with a null attachment. Let me remove this unneeded check.